dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ObjectReport.java

```diff
@@ -226,7 +227,8 @@ public boolean hasErrorReport( Predicate<ErrorReport> test )
     @Override
     public void forEachErrorReport( Consumer<ErrorReport> reportConsumer )
     {
-        errorReportsByCode.values().forEach( reports -> reports.forEach( reportConsumer ) );
+        errorReportsByCode.values().stream().flatMap( errors -> errors.stream() )
+            .collect( Collectors.toList() ).forEach( errorReport -> reportConsumer.accept( errorReport ) );
```

Comment:
`.collect( Collectors.toList() ).forEach(` appears unnecessarily creating a intermediate collection just to enter stream processing again.